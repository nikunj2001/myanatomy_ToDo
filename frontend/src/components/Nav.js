import React from 'react'
import "./style.css"
const Nav = () => {
  return (
    <div className='nav'>
        <ul>
            <li>
                To-Do List
            </li>
        </ul>
    </div>
  )
}

export default Nav