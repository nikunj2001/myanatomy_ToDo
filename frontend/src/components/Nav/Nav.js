import React from 'react'
import "../style.css"
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { userLogout } from '../../store/asyncMethods/AuthMethods';
const Nav = () => {
  const dispatch = useDispatch();
  const { user } = useSelector(state => state.AuthReducer);
  const handleLogout = e => {
    dispatch(userLogout());
  }
  return (
    <div className='nav'>
      <div className="nav-left">
        <ul>
          <li>To-Do List</li>
        </ul>
      </div>
      <div className='nav-right' >
        <ul>
          {
            user ? (
              <>
                <li>
                  {user.name}
                </li>
                <li onClick={handleLogout} >
                  Logout
                </li>
              </>
            ) :
              (
                <>
                  <Link to="/" >Login</Link>
                  <Link to='/register' >Register</Link>
                </>
              )
          }

        </ul>
      </div>
    </div>
  )
}

export default Nav