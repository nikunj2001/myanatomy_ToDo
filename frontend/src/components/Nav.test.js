import React from "react";
import renderer from "react-test-renderer";
import Nav from "./Nav";

it('has two components', () => {
  const tree = renderer.create(<Nav/>).toJSON();
  expect(tree).toMatchSnapshot();
});
