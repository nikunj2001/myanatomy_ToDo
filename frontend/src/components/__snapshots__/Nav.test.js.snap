// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has two components 1`] = `
<div
  className="nav"
>
  <ul>
    <li>
      To-Do List
    </li>
  </ul>
</div>
`;
